sky/backends/backend_utils.py

```diff
@@ -798,7 +799,12 @@ def write_cluster_config(
     excluded_clouds = []
     remote_identity = skypilot_config.get_nested(
         (str(cloud).lower(), 'remote_identity'), 'LOCAL_CREDENTIALS')
-    if remote_identity == 'SERVICE_ACCOUNT':
+    if remote_identity is not None and not isinstance(remote_identity, str):
+        for profile in remote_identity:
+            if fnmatch.fnmatchcase(cluster_name_on_cloud, profile):
+                remote_identity = remote_identity[profile]
+                break
+    if remote_identity != 'LOCAL_CREDENTIALS':
```

Comment:
Does this make sense for all cases when `LOCAL_CREDENTIALS` is not specified?

Do we need to update the schema for the `remote_identity` to be more like:
```yaml
aws:
  remote_identity:
    sky-serve-controller-*:
      name: skypilot-fake
      type: SERVICE_ACCOUNT
    "*":
      name: skypilot-fake-default
      type: LOCAL_CREDENTIALS
```

Reply:
Yes, the current implementation seems good to me. 

For the proposed schema, I am thinking it would be nice to keep it the current way just for simplicity. Since we don't actually support automatically switching between multiple user account locally at the moment.